tests/backup/backup_rancher_test.go

```diff
@@ -1,6 +1,192 @@
+			restoreName := fmt.Sprintf("%s-same-project-%v", restoreNamePrefix, RandomString(10))
+			restoreList = append(restoreList, restoreName)
+			err = CreateRestore(restoreName, backupName, namespaceMapping, SourceClusterName, orgID, ctx, nil)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creating restote- %s from backup %s", restoreName, backupName))
+		})
+
+		Step("Restoring to different project and different namespace in same cluster", func() {
+			var restoredNamespaceList []string
+			log.InfoD("Restoring to different project and different namespace in same cluster")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range appNamespaces {
+				restoredNamespace := "restored-different-project-" + RandomString(5)
+				namespaceMapping[namespace] = restoredNamespace
+				restoredNamespaceList = append(restoredNamespaceList, restoredNamespace)
+				restoreNamespacesAll = append(restoreNamespacesAll, restoredNamespace)
+				nsObject := &corev1.Namespace{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: restoredNamespace,
+					},
+				}
+				_, err = core.Instance().CreateNamespace(nsObject)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating namespace %s", restoredNamespace))
+			}
+			err = Inst().S.(*rke.Rancher).AddNamespacesToProject(sourceClusterProjectList[1], restoredNamespaceList)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Adding namespaces %s to project %s", restoredNamespaceList, sourceClusterProjectList[1]))
+			err = Inst().S.(*rke.Rancher).ValidateProjectOfNamespaces(sourceClusterProjectList[1], restoredNamespaceList)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("verifying project %s of namespace %s", sourceClusterProjectList[1], restoredNamespaceList))
+			restoreName := fmt.Sprintf("%s-diff-project-%v", restoreNamePrefix, RandomString(10))
+			restoreList = append(restoreList, restoreName)
+			err = CreateRestore(restoreName, backupName, namespaceMapping, SourceClusterName, orgID, ctx, nil)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creating restore- %s from backup %s", restoreName, backupName))
+			err = Inst().S.(*rke.Rancher).ValidateProjectOfNamespaces(sourceClusterProjectList[1], restoredNamespaceList)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying project %s of namespace %s", sourceClusterProjectList[1], restoredNamespaceList))
+		})
+	})
+
+	JustAfterEach(func() {
+		defer EndPxBackupTorpedoTest(scheduledAppContexts)
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		opts := make(map[string]bool)
+		opts[SkipClusterScopedObjects] = true
+		DestroyApps(scheduledAppContexts, opts)
+		for _, ns := range restoreNamespacesAll {
+			err = core.Instance().DeleteNamespace(ns)
+			log.FailOnError(err, "Deletion of namespace %s failed", ns)
+		}
+		for _, project := range sourceClusterProjectList {
+			Inst().S.(*rke.Rancher).DeleteRancherProject(project, "local")
```

Comment:
How about checking if `Inst().S` is of type `*rke.Rancher` before using an assertion?

I think you can use a switch statement like `switch s := Inst().S.(type)` and include a case for `*rke.Rancher`.

Reply:
This testcase will only run on Rancher cluster.